src/main/java/com/linecorp/armeria/common/SerializationFormat.java

```diff
@@ -62,6 +82,18 @@
     public static Set<SerializationFormat> ofThrift() {
         return THRIFT_FORMAT;
     }
+
+    /**
+     * Returns the serialization format corresponding to the passed in {@code mimeType}, or
+     * {@link Optional#empty} if the mimetype is not recognized. {@code null} is treated as an unknown
+     * mimetype.
+     */
+    public static Optional<SerializationFormat> fromMimeType(@Nullable String mimeType) {
+        if (mimeType == null) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(MIME_TYPE_TO_FORMAT.get(MIME_TYPE_NORMALIZER.apply(mimeType)));
```

Comment:
This won't work when there are more than one parameter. Perhaps we could...

- Split by semicolon
- Use `javax.activation.MimeTypeParameterList` to parse the parameters? (It's part of JRE 1.8)